### PR TITLE
feat_: add KeycardPairingKey

### PR DESCRIPTION
**Pull request by flexsurfer**
On mobile we don't have KeycardPairing json File, so we are unable to provide a path to it, so for recoveryAccount we'll pass a parameter KeycardPairingKey which will be used instead of KeycardPairing json File, on desktop it will continue work as it is

### CODE REVIEW

**Review by igor-sirotin — COMMENTED** [review]
1. Can you please add some PR description?
In which case and why do we need to pass `KeycardPairingKey`?

2. I'm unsure how it will work further without the `KeycardPairingDataFile`?

**Comment by flexsurfer** [comment]
thank @igor-sirotin, i've added the description, for 2) on mobile it's not needed because we don't have sync implemented, for desktop it will work fine.

File: protocol/requests/create_account.go [hunk 1]
```diff
@@ -80,6 +80,7 @@ type CreateAccount struct {
 	APIConfig *APIConfig `json:"apiConfig"`
 
 	KeycardInstanceUID     string  `json:"keycardInstanceUID"`
+	KeycardPairingKey      string  `json:"keycardPairingKey"`
```
Comment (igor-sirotin) [inline protocol/requests/create_account.go:86]: Can you please mention here that only `KeycardPairingKey` can be used  instead and has precedence over `KeycardPairingKey`?

File: api/geth_backend.go [hunk 2]
```diff
@@ -1713,24 +1713,31 @@ func (b *GethStatusBackend) prepareForKeycard(request *requests.CreateAccount, i
...
+		keycard, ok := pairings[request.KeycardInstanceUID]
+		if !ok {
+			return nil, errors.New("keycard not found in pairings file")
+		}
+
+		response.settings.KeycardPairing = keycard.Key
+		response.account.KeycardPairing = keycard.Key
+	} else {
+		response.settings.KeycardPairing = request.KeycardPairingKey
+		response.account.KeycardPairing = request.KeycardPairingKey
```
Comment (igor-sirotin) [inline api/geth_backend.go]: Would be nice to invert this condition to have the shorter branch to go first.

```suggestion
	if request.KeycardPairingKey != "" {
		response.settings.KeycardPairing = request.KeycardPairingKey
		response.account.KeycardPairing = request.KeycardPairingKey
	} else {
		kp := wallet.NewKeycardPairings()
		kp.SetKeycardPairingsFile(response.nodeConfig.KeycardPairingDataFile)
		pairings, err := kp.GetPairings()
		if err != nil {
			return nil, errors.Wrap(err, "failed to get keycard pairings")
		}

		keycard, ok := pairings[request.KeycardInstanceUID]
		if !ok {
			return nil, errors.New("keycard not found in pairings file")
		}

		response.settings.KeycardPairing = keycard.Key
		response.account.KeycardPairing = keycard.Key
	}
```

**Review by igor-sirotin — APPROVED** [review]
I guess this makes sense 👍 
Though it's yet another brick in the wall of complexity, we need to simplify this keycard pairing file work

**Review by saledjenic — APPROVED** [review]
I don't want to block this PR since it works in the mobile app and doesn't break the desktop app, but I think it collides a bit with the intention from this PR https://github.com/status-im/status-go/pull/4014 and it won't work when syncing or importing keypairs from another device either that's done between 2 mobile instances or mobile and desktop (desktop -> desktop should work).

**Comment by flexsurfer** [comment]
thank you @saledjenic i will create a task for mobile so we could research and improve that

**Pull request merged**